rs/nns/governance/src/voting.rs

```diff
@@ -0,0 +1,432 @@
+use crate::{
+    neuron_store::NeuronStore,
+    pb::v1::{Ballot, Topic, Topic::NeuronManagement, Vote},
+};
+use ic_nns_common::pb::v1::{NeuronId, ProposalId};
+use std::{
+    cell::RefCell,
+    collections::{BTreeMap, BTreeSet, HashMap},
+};
+
+thread_local! {
+    static VOTING_STATE_MACHINES: RefCell<VotingStateMachines> = RefCell::new(VotingStateMachines::new());
+}
+
+struct VotingStateMachines {
+    // machines
```

Comment:
The comment here seems not very useful. Perhaps you can explain the rationale of keying by proposal id. Or removing it is fine, since it's somewhat explained in cast_vote_and_cascade_follow